### PR TITLE
feat: Upgrade to Phoenix 1.7

### PR DESCRIPTION
**Pull request by nbw**
## What kind of change does this PR introduce?

Upgrade phoenix to 1.7.0. **Is it needed?** maybe. The upgrade to Phoenix 1.7 was quite a chore depending on your app so it's good to do sooner than later.

I followed the CHANGELOG and guide. The commits should be a good indicator as to the reasoning of each change. 

- https://github.com/phoenixframework/phoenix/blob/main/CHANGELOG.md
- https://gist.github.com/chrismccord/00a6ea2a96bc57df0cce526bd20af8a7

## Merge it?

It requires some testing. This PR could serve as a guide for the maintainers to do the upgrade themselves. 

## Additional context

- skipped [Migrate Phoenix.View to Phoenix.Component](https://gist.github.com/chrismccord/00a6ea2a96bc57df0cce526bd20af8a7#optional-migrate-from-phoenixview-to-phoenixcomponent) in the guide

- Couldn't get all tests (about 8 failures) to pass locally because it requires a setup that I'm not familiar with (something multitenant?).

- Also this section of _seeds_after_migration_ wasn't working for me:

```
{:ok, _} = Repo.transaction(fn ->
  [
    "drop table if exists \"public\".\"test\";",
    "create sequence if not exists test_id_seq;",
    "create table \"public\".\"test\" (
        \"id\" int4 not null default nextval('test_id_seq'::regclass),
        \"details\" text,
        primary key (\"id\")
    );",
    "grant all on table public.test to anon;",
    "grant all on table public.test to postgres;",
    "grant all on table public.test to authenticated;",
    "create publication #{publication} for all tables"
  ] |> Enum.each(&query(Repo, &1, []))
end)
```

### CODE REVIEW

**Review by zezo2019 — APPROVED** [review]

**Comment by w3b6x9** [comment]
@nbw thanks for your contribution! can you please run `mix format` and push that up?

**Comment by nbw** [comment]
@w3b6x9 done! If there's a github way to let others contribute to the fork then let me know (I couldn't find anything)

File: lib/realtime_web.ex [hunk 1]
```diff
@@ -111,4 +115,15 @@ defmodule RealtimeWeb do
...
+  def verified_routes do
+    quote do
+      use Phoenix.VerifiedRoutes,
+        endpoint: RealtimeWeb.Endpoint,
+        router: RealtimeWeb.Router,
+        statics: RealtimeWeb.static_paths()
+    end
+  end
+
+  def static_paths, do: ~w(assets fonts images favicon.ico robots.txt)
```
Comment (w3b6x9) [inline lib/realtime_web.ex:128]: @nbw nitpick: can you please move `static_paths/0` to the very top?

File: lib/realtime_web.ex [hunk 2]
```diff
@@ -39,13 +41,15 @@ defmodule RealtimeWeb do
 
       # Include shared imports and aliases for views
       unquote(view_helpers())
+
+      unquote(verified_routes())
```
Comment (w3b6x9) [inline lib/realtime_web.ex:45]: @nbw i think this is supposed to go in `view_helpers/0` according to https://gist.github.com/chrismccord/00a6ea2a96bc57df0cce526bd20af8a7#optional-update-your-app-to-support-phoenixverifiedroutes.

**Comment by w3b6x9** [comment]
> If there's a github way to let others contribute to the fork then let me know (I couldn't find anything)

@nbw yea i'm not sure either. normally i would create a branch in this repo, merge in the branch from fork, and work off of that.

definitely open to other ways of doing it.

**Comment by w3b6x9** [comment]
@nbw made a few more comments but lgtm!

**Comment by w3b6x9** [comment]
@chasers @abc3 i tested this on staging and it lgtm.

**Comment by w3b6x9** [comment]
@supabase/realtime remember to version bump in `mix.exs` once this is merged into https://github.com/supabase/realtime/tree/chore/phoenix-1.7

File: lib/realtime_web.ex (hunk 2 above)
Comment (w3b6x9) [reply lib/realtime_web.ex]: @nbw thanks! yep, i tested it on our staging and it's looking good.

**Review by w3b6x9 — APPROVED** [review]
@nbw lgtm! thanks!

**Pull request merged**